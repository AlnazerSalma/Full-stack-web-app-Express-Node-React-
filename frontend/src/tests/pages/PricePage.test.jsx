import React from "react";
import { render, screen } from "@testing-library/react";
import PricePage from "../../pages/PricingPage"; 
import '@testing-library/jest-dom';
import mockPricing from "../__mocks_data__/mockPricing";
import { mockUseFetch } from "../__test_utils__/mockHooks";


describe("PricePage", () => {
  afterEach(() => {
    jest.clearAllMocks(); // Clear mocks after each test to avoid leakage
  });

  it("displays loading state", () => {
    mockUseFetch({
      loading: "Loading...",
    });
    render(<PricePage />);
    expect(screen.getByText("Loading...")).toBeInTheDocument();
  });

  it("displays error message", () => {
    mockUseFetch({
      error: "Failed to fetch",
    });

    render(<PricePage />);
    expect(screen.getByText("Failed to fetch")).toBeInTheDocument();
  });

  it("displays pricing cards when data is available", () => {
    mockUseFetch({
      data: [mockPricing[0]],
    });

    render(<PricePage />);

    expect(screen.getByText("Pro")).toBeInTheDocument();
    expect(screen.getByText("Pro plan description")).toBeInTheDocument();
    expect(screen.getByText("$19")).toBeInTheDocument();
    expect(screen.getByText("Feature 1")).toBeInTheDocument();
    expect(screen.getByText("Feature 3")).toBeInTheDocument();
  });

  it("matches snapshot with all pricing plans", () => {
    mockUseFetch({
      data: mockPricing, 
    });

    const { asFragment } = render(<PricePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
