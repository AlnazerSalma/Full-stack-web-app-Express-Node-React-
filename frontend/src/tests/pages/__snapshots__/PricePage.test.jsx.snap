// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricePage renders PricePage with pricing cards correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="project-section container-fluid"
  >
    <div
      class="container"
    >
      <h1
        class="project-heading"
      >
        Pricing & Contact
      </h1>
      <div
        class="row"
        style="justify-content: center; padding-bottom: 10px;"
      >
        <div
          class="project-card col-lg-4 col-md-6 col-12"
        >
          <div
            class="project-card-view p-3 d-flex flex-column card"
          >
            <div
              class="d-flex align-items-center justify-content-between mb-3"
            >
              <img
                alt="card-img"
                class="card-small-img"
                src="/img/pro.png"
              />
              <div
                class="subscribe-badge"
              >
                Subscribe
              </div>
            </div>
            <div
              class="flex-grow-1 d-flex flex-column"
            >
              <div
                class="text-left"
              >
                <div
                  class="card-title card-title h5"
                >
                  Pro
                </div>
                <p
                  class="card-text card-text"
                >
                  Pro plan description
                </p>
              </div>
              <div
                class="card-price mt-auto"
              >
                <span>
                  $19
                </span>
                <span
                  class="billing-cycle"
                >
                  / month
                </span>
              </div>
            </div>
            <div
              class="mt-3 text-start"
            >
              <a
                class="subscribe-badge text-decoration-none d-inline-block subscribe-btn"
                href="https://example.com/demo-pro"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="margin-bottom: 3px;"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M14 7C13.4477 7 13 7.44772 13 8V16C13 16.5523 13.4477 17 14 17H18C18.5523 17 19 16.5523 19 16V8C19 7.44772 18.5523 7 18 7H14ZM17 9H15V15H17V9Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M6 7C5.44772 7 5 7.44772 5 8C5 8.55228 5.44772 9 6 9H10C10.5523 9 11 8.55228 11 8C11 7.44772 10.5523 7 10 7H6Z"
                    fill="currentColor"
                  />
                  <path
                    d="M6 11C5.44772 11 5 11.4477 5 12C5 12.5523 5.44772 13 6 13H10C10.5523 13 11 12.5523 11 12C11 11.4477 10.5523 11 10 11H6Z"
                    fill="currentColor"
                  />
                  <path
                    d="M5 16C5 15.4477 5.44772 15 6 15H10C10.5523 15 11 15.4477 11 16C11 16.5523 10.5523 17 10 17H6C5.44772 17 5 16.5523 5 16Z"
                    fill="currentColor"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M4 3C2.34315 3 1 4.34315 1 6V18C1 19.6569 2.34315 21 4 21H20C21.6569 21 23 19.6569 23 18V6C23 4.34315 21.6569 3 20 3H4ZM20 5H4C3.44772 5 3 5.44772 3 6V18C3 18.5523 3.44772 19 4 19H20C20.5523 19 21 18.5523 21 18V6C21 5.44772 20.5523 5 20 5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
                   Book a call
              </a>
            </div>
            <div
              class="gray-line"
            />
            <div
              class="card-row row"
            >
              <div
                class="col-sm-6"
              >
                <div
                  class="card-column"
                >
                  <p>
                    Feature 1
                  </p>
                  <p>
                    Feature 2
                  </p>
                </div>
              </div>
              <div
                class="col-sm-6"
              >
                <div
                  class="card-column"
                >
                  <p>
                    Feature 3
                  </p>
                  <p>
                    Feature 4
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="project-card col-lg-4 col-md-6 col-12"
        >
          <div
            class="project-card-view p-3 d-flex flex-column card"
          >
            <div
              class="d-flex align-items-center justify-content-between mb-3"
            >
              <img
                alt="card-img"
                class="card-small-img"
                src="/img/basic.png"
              />
              <div
                class="subscribe-badge"
              >
                Subscribe
              </div>
            </div>
            <div
              class="flex-grow-1 d-flex flex-column"
            >
              <div
                class="text-left"
              >
                <div
                  class="card-title card-title h5"
                >
                  Basic
                </div>
                <p
                  class="card-text card-text"
                >
                  Basic plan description
                </p>
              </div>
              <div
                class="card-price mt-auto"
              >
                <span>
                  $9
                </span>
                <span
                  class="billing-cycle"
                >
                  / month
                </span>
              </div>
            </div>
            <div
              class="mt-3 text-start"
            >
              <a
                class="subscribe-badge text-decoration-none d-inline-block subscribe-btn"
                href="https://example.com/demo"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="margin-bottom: 3px;"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M14 7C13.4477 7 13 7.44772 13 8V16C13 16.5523 13.4477 17 14 17H18C18.5523 17 19 16.5523 19 16V8C19 7.44772 18.5523 7 18 7H14ZM17 9H15V15H17V9Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M6 7C5.44772 7 5 7.44772 5 8C5 8.55228 5.44772 9 6 9H10C10.5523 9 11 8.55228 11 8C11 7.44772 10.5523 7 10 7H6Z"
                    fill="currentColor"
                  />
                  <path
                    d="M6 11C5.44772 11 5 11.4477 5 12C5 12.5523 5.44772 13 6 13H10C10.5523 13 11 12.5523 11 12C11 11.4477 10.5523 11 10 11H6Z"
                    fill="currentColor"
                  />
                  <path
                    d="M5 16C5 15.4477 5.44772 15 6 15H10C10.5523 15 11 15.4477 11 16C11 16.5523 10.5523 17 10 17H6C5.44772 17 5 16.5523 5 16Z"
                    fill="currentColor"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M4 3C2.34315 3 1 4.34315 1 6V18C1 19.6569 2.34315 21 4 21H20C21.6569 21 23 19.6569 23 18V6C23 4.34315 21.6569 3 20 3H4ZM20 5H4C3.44772 5 3 5.44772 3 6V18C3 18.5523 3.44772 19 4 19H20C20.5523 19 21 18.5523 21 18V6C21 5.44772 20.5523 5 20 5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
                   Book a call
              </a>
            </div>
            <div
              class="gray-line"
            />
            <div
              class="card-row row"
            >
              <div
                class="col-sm-6"
              >
                <div
                  class="card-column"
                >
                  <p>
                    Feature 1
                  </p>
                  <p>
                    Feature 2
                  </p>
                </div>
              </div>
              <div
                class="col-sm-6"
              >
                <div
                  class="card-column"
                >
                  <p>
                    Feature 3
                  </p>
                  <p>
                    Feature 4
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
